R/estimateDiversity.R

```diff
@@ -283,19 +288,26 @@ setMethod("estimateDiversity", signature = c(x="TreeSummarizedExperiment"),
             name = index, tree_name = "phylo", 
             ..., BPPARAM = SerialParam()){
         # input check
+        supported_index <- c("coverage", "fisher", "gini_simpson", "faith",
+                             "inverse_simpson", "log_modulo_skewness", "shannon")
+        index_string <- paste0("'", paste0(supported_index, collapse = "', '"), "'")
+        if ( !all(index %in% supported_index) || length(index) == 0) {
+            stop(paste("Unsupported index:", index_string), call. = FALSE)
```

Comment:
The error message could still be more informative.

"'index' must be from the following options:"